### ISSUE TITLE
Documentation: breadcrumbs to available shapes for various geoms?

### ISSUE THREAD

**Issue by jwhendy** [has2k1/plotnine#411]
Greetings,

I was looking for how to specify the shape, but had a hard time finding the list of all of them. R's `ggplot2` includes a nice reference of all of them at the bottom of their [`scale_shape` docs](https://ggplot2.tidyverse.org/reference/scale_shape.html).

![image](https://user-images.githubusercontent.com/379357/86372446-c6122800-bc47-11ea-82f5-87f98a3dae51.png)

In addition, it doesn't really matter as you can just guess because they are numeric. I could see that `plotnine` uses some type of string specifier via the `geom_point` docs, but no link or reference to what the other acceptable string values were?

![grab_2020-07-02_093846](https://user-images.githubusercontent.com/379357/86372496-d9bd8e80-bc47-11ea-8e7d-11a6a9870ffb.png)

In searching all the scale_shape variants, I struck out. When I specifically searched `shape 'o'`, I found it in [the source](https://plotnine.readthedocs.io/en/stable/_modules/plotnine/scales/scale_shape.html).

Could this handy table be added to, perhaps `scale_shape` for reference?

---

Related: this would be a breaking change, but any thoughts on cleaning up/standardizing the definitions? I'm guessing this is tied into something else which defines them and that's a long shot request...

I admit to a bit of cringe when I saw that we have both `str: '1', '2', '3', '4'` as well as `int: 1, 2, 3, 4` in the shape specifiers.

**Comment by has2k1** [comment]
> I admit to a bit of cringe when I saw that we have both str: '1', '2', '3', '4' as well as int: 1, 2, 3, 4 in the shape specifiers.

Yes that is cringe-worthy, but that is the API are the values provided by Matplotlib our backend. 

I think the solution is to create a visual reference in the documentation.

**Comment by jwhendy** [comment]
I figured this :) 

Would you like me to create such a visual reference? Would a table like the ggplot one be your preference, or something else? I could just do a table matching the source so it'd be two mini tables, one for filled and unfilled? or perhaps side by side, so two columns each with a gap in between?

If you had a reference for where those images live, it could make life easier, otherwise I'll likely iterate through this with big sizes and auto-screenshots.

**Comment by has2k1** [comment]
Matplotlib has a table of markers at https://matplotlib.org/3.1.1/api/markers_api.html. We can have a similar table. I looked at the MPL code and they do not include the script (for us to copy) that generated the images. I want the code that generates the images to be a part of the repository.

If this is too much I can do it.

**Comment by jwhendy** [comment]
That said, what do you think about this?
- symbol names [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/markers.py) are synonymous with the image names [here](view-source:https://matplotlib.org/3.1.1/api/markers_api.html) (source for the same page you found). Those link to `doc/_static/markers/symbol_name.png`, which correspond.

Could we pull from the table [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/markers.py) to use for this?

**Comment by has2k1** [comment]
I want to create our own images for the symbols so that the script that creates them is part of the repository.

**Comment by jwhendy** [comment]
Would we be replicating [markers.py](https://github.com/matplotlib/matplotlib/blob/224e45186e9253fd4b063961153c38f497300f2d/lib/matplotlib/markers.py) where they define the drawing functions, then?

It appears they use their own [path library](https://github.com/matplotlib/matplotlib/blob/224e45186e9253fd4b063961153c38f497300f2d/lib/matplotlib/path.py) to draw these. For example:

```
_caret_path = Path([[-1.0, 1.5], [0.0, 0.0], [1.0, 1.5]])

    def _set_caretdown(self):
        self._transform = Affine2D().scale(0.5)
        self._snap_threshold = 3.0
        self._filled = False
        self._path = self._caret_path
        self._joinstyle = 'miter'
```
I'm a out of my python league, I fear... it's not clear to me how a `marker=foo` in a call would trace back to `self._set_caretdown()`. I'm not finding that in the code, so I don't know when these ever get called? Ok, actually perhaps it's this?

```
elif (not isinstance(marker, (np.ndarray, list)) and
              marker in self.markers):
            self._marker_function = getattr(
                self, '_set_' + self.markers[marker])
```

Sorry if you already saw this... I was hunting around to understand a bit more how they do it given your comment:

> I looked at the MPL code and they do not include the script (for us to copy) that generated the images.

Here is another page that draws out each shape, and that traces back to actual code, which maybe is helpful?
- the code iterates through sets of unfilled/filled markers
- those are defined [here](https://github.com/matplotlib/matplotlib/blob/8b9f61d001a363c74a2832f8e7c435b6b59c176b/examples/lines_bars_and_markers/marker_reference.py#L42)
- and those are subsequently defined [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/markers.py#L159)

I'm still a bit unclear on how the magic happens to get them plotted.

Sorry if I'm not actually being helpful. I enjoy trying to figure these things out, though admittedly my level of coding is at the data-analysis/plotting level, with little/no experience in understanding how bigger projects like this are structured! I appreciate the patience/coaching, and am happy to help if I'm able!

**Comment by has2k1** [comment]
I did not mean to actually code the paths that draw the objects. I want to use Matplotlib and very small square axes (without any axis or tick mark) and plot each symbol and save. If it is still confusing, no worries I will do it.
